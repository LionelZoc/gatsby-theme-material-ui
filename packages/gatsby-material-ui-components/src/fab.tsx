import MuiFab, { FabProps } from "@material-ui/core/Fab";

import patchButtonBaseComponent from "./patch-base-button-components";

export const Fab = patchButtonBaseComponent<FabProps>(MuiFab);
