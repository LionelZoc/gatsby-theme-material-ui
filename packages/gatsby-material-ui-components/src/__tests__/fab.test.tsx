import * as React from 'react';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { Fab } from '../fab';

describe('fab', () => {
  beforeEach(() => {
    // gatsby link unit tests crashed when globals are undefined
    (global as any).___loader = {
      enqueue: jest.fn(),
    };
  });

  it('should be rendered correctly', () => {
    const { asFragment } = render(<Fab>text</Fab>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('lbutton with to', () => {
    const { asFragment } = render(<Fab to="/page2">text</Fab>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('button with ref', () => {
    const { result } = renderHook(() => React.useRef<HTMLButtonElement>(null));
    const buttonRef = result.current;
    const { asFragment } = render(<Fab ref={buttonRef}>text</Fab>);
    expect(asFragment()).toMatchSnapshot();
  });
});
