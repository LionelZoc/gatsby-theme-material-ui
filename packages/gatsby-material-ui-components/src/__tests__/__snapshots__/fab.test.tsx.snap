// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fab button with ref 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiFab-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      text
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`fab lbutton with to 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiFab-root"
    href="/page2"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiFab-label"
    >
      text
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</DocumentFragment>
`;

exports[`fab should be rendered correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiFab-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      text
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
